apps/ios/Shared/Views/NewChat/NewChatView.swift

```diff
@@ -0,0 +1,959 @@
+                id: "processQRCode: failure"
+            ))
+        }
+    }
+
+    private func connect(_ link: String) {
+        planAndConnect(
+            link,
+            showAlert: { alert = .planAndConnectAlert(alert: $0) },
+            showActionSheet: { sheet = $0 },
+            dismiss: true,
+            incognito: nil
+        )
+    }
+}
+
+private func linkTextView(_ link: String) -> some View {
+    Text(link)
+        .lineLimit(1)
+        .font(.caption)
+        .truncationMode(.middle)
+}
+
+struct InfoSheetButton<Content: View>: View {
+    @ViewBuilder let content: Content
+    @State private var showInfoSheet = false
+
+    var body: some View {
+        Button {
+            showInfoSheet = true
+        } label: {
+            Image(systemName: "info.circle")
+                .resizable()
+                .scaledToFit()
+                .frame(width: 24, height: 24)
+        }
+        .sheet(isPresented: $showInfoSheet) {
+            content
+        }
+    }
+}
+
+func strIsSimplexLink(_ str: String) -> Bool {
+    if let parsedMd = parseSimpleXMarkdown(str),
+       parsedMd.count == 1,
+       case .simplexLink = parsedMd[0].format {
+        return true
+    } else {
+        return false
+    }
+}
+
+func strHasSingleSimplexLink(_ str: String) -> FormattedText? {
+    if let parsedMd = parseSimpleXMarkdown(str) {
+       let parsedLinks = parsedMd.filter({ $0.format?.isSimplexLink ?? false })
+        if parsedLinks.count == 1 {
+            return parsedLinks[0]
+        } else {
+            return nil
+        }
+    } else {
+        return nil
+    }
+}
+
+struct IncognitoToggle: View {
+    @Binding var incognitoEnabled: Bool
+    @State private var showIncognitoSheet = false
+
+    var body: some View {
+        ZStack(alignment: .leading) {
+            Image(systemName: incognitoEnabled ? "theatermasks.fill" : "theatermasks")
+                .frame(maxWidth: 24, maxHeight: 24, alignment: .center)
+                .foregroundColor(incognitoEnabled ? Color.indigo : .secondary)
+                .font(.system(size: 14))
+            Toggle(isOn: $incognitoEnabled) {
+                HStack(spacing: 6) {
+                    Text("Incognito")
+                    Image(systemName: "info.circle")
+                        .foregroundColor(.accentColor)
+                        .font(.system(size: 14))
+                }
+                .onTapGesture {
+                    showIncognitoSheet = true
+                }
+            }
+            .padding(.leading, 36)
+        }
+        .sheet(isPresented: $showIncognitoSheet) {
+            IncognitoHelp()
+        }
+    }
+}
+
+func sharedProfileInfo(_ incognito: Bool) -> Text {
+    let name = ChatModel.shared.currentUser?.displayName ?? ""
+    return Text(
+        incognito
+        ? "A new random profile will be shared."
+        : "Your profile **\(name)** will be shared."
+    )
+}
+
+enum PlanAndConnectAlert: Identifiable {
```

Comment:
Everything below is "plan and connect" logic moved from NewChatButton. There are minor changes for cleanup and search logic:
- optional `cleanup` functions - used in NewChat "Connect via link" tab and in chat list search to remove pasted link from the view. They're called:
  - wherever connection is cancelled via sheet or alert,
  - or connection is in progress (e.g. invitationLinkConnecting plan),
  - or after connection is started (e.g. after apiConnect in connectViaLink).
- optional `filterKnownContact` and `filterKnownGroup` functions - used in chat list search to filter a "known" chat returned by plan. If they are not null and plan returns a "known" contact or group, they're called instead of immediately opening conversations.